File: javascript/distortion_correct_token_demo.js
/**
 * 扭曲校正服务token 方式请求的使用示例
 */
var discor = require("./moderation_sdk/distortion_correct");
var token = require("./moderation_sdk/gettoken");
var utils = require("./moderation_sdk/utils");

// 初始化服务的区域信息，目前支持华北-北京(cn-north-4)区域信息
utils.initRegion("cn-north-4");

var username = "*******";        // 配置用户名
var domain_name = "*******";     // 配置用户名
var password = "*******";        // 密码

var filepath = "./data/moderation-distortion.jpg";
var data = utils.changeFileToBase64(filepath);

// obs链接需要和region区域一致，不同的region的obs资源不共享
demo_data_url = "https://sdk-obs-source-save.obs.cn-north-4.myhuaweicloud.com/vat-invoice.jpg";

token.getToken(username, domain_name, password, function (token) {

    discor.distortion_correct(token, data, "", true, function (result) {
        var resultObj = JSON.parse(result);

        if (resultObj.result.data !== "") {
            utils.getFileByBase64Str("./data/modeation_distortion-token-1.jpg", resultObj.result.data);
        }else{
            console.log(result);
        }
    });

    discor.distortion_correct(token, "", demo_data_url, true, function (result) {
        var resultObj = JSON.parse(result);

        if (resultObj.result.data !== "") {
            utils.getFileByBase64Str("./data/modeation_distortion-token-2.jpg", resultObj.result.data);
        }else{
            console.log(result);
        }
    });
});

File: python/distortion_correct_token_demo.py
# -*- coding:utf-8 -*-
from moderation_sdk.gettoken import get_token
from moderation_sdk.utils import encode_to_base64
from moderation_sdk.utils import decode_to_wave_file
from moderation_sdk.distortion_correct import distortion_correct
from moderation_sdk.utils import init_global_env
import json

if __name__ == '__main__':
    # Services currently support North China-Beijing(cn-north-4)
    init_global_env('cn-north-4')

    #
    # access moderation distortion correct post data by token
    #
    user_name = '******'
    password = '******'
    account_name = '******'  # the same as user_name in commonly use

    demo_data_url = 'https://sdk-obs-source-save.obs.cn-north-4.myhuaweicloud.com/modeation-distortion.jpg'
    token = get_token(user_name, password, account_name)

    #call interface use the url correction is true means do not correction
    result = distortion_correct(token, '', demo_data_url, True)
    result_obj = json.loads(result)

    # If the image needs to be corrected, the corrected image will be generated in the data directory,
    # otherwise the processing result will be printed.
    if result_obj['result']['data'] !='':
        decode_to_wave_file(result_obj['result']['data'], 'data/modeation-distortion-token-1.png')
    else:
        print(result)

    # call interface use the file
    result = distortion_correct(token, encode_to_base64('data/modeation-distortion.jpg'), '', True)
    result_obj = json.loads(result)

    # If the image needs to be corrected, the corrected image will be generated in the data directory,
    # otherwise the processing result will be printed.
    if result_obj['result']['data'] != '':
        decode_to_wave_file(result_obj['result']['data'], 'data/modeation-distortion-token-2.png')
    else:
        print(result)
